string checker
uint64[] buffer_p
uint64[] buffer_o
uint32[] timerBounds
float64[] timerInstances
bool[] timerCache
uint64[] currAss
uint64[] nextAss
uint64[] currAnt
uint64[] nextAnt
uint32 conIns
uint32 endIns
uint32 ATCF
bool[] last_p
time last_msg_ts
